File: 1047. 删除字符串中的所有相邻重复项/1047. 删除字符串中的所有相邻重复项.py
class Solution:
    def removeDuplicates(self, S):
        mystack = []
        res = ''
        for c in S:
            if not mystack:
                mystack.append(c)
            elif mystack and mystack[-1] != c:
                mystack.append(c)
            elif mystack[-1] == c:
                mystack.pop(-1)
        return ''.join(mystack)

s = Solution()
t = s.removeDuplicates("abbacds")
print(t)

File: 1047. 删除字符串中的所有相邻重复项/1047. 删除字符串中的所有相邻重复项.go
package main

import "fmt"

func main() {
	fmt.Println(removeDuplicates("abbaca"))
}

func removeDuplicates(S string) string {
	mystack := make([]rune, 0, 20000)
	for i, c := range S {
		// TODO
		fmt.Println(i, mystack)
	}
	res := ""
	return res
}
